src/validation_docker_workflow/pull_docker_image.py

```diff
@@ -0,0 +1,48 @@
+# Copyright OpenSearch Contributors
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import logging
+import subprocess
+from subprocess import PIPE
+from typing import Any
+
+"""
+This class is to pull the artifact from the dockerHub.
+If the image is currently existing at local, we want to remove it and download a fresh one as the artifact may have been renewed at hub.
+"""
+
+
+class PullDockerImage():
+
+    @classmethod
+    def pull_image(self, image_name: str) -> Any:
+        local_inspect = "docker image inspect -f '{{ .Id }}' " + image_name
+        local_remove = "docker image rm -f " + image_name
+        dockerHub_pull = "docker pull " + image_name
+
+        # Removing the image if it exists at local
+        result_inspect = subprocess.run(local_inspect, shell=True, stdout=PIPE, stderr=PIPE, universal_newlines=True)
+        if (result_inspect.returncode == 0):
```

Comment:
how about we use try/catch to handle errors?

Reply:
Sounds good !! Let me do that ... thanks for the input!! @rishabh6788